### PR TITLE
Add tests and types for Polygon & earcut

### PR DESCRIPTION
**Pull request by felixpalmer**
Addresses missing tests from https://github.com/uber-web/math.gl/issues/220

### CODE REVIEW

**Comment by felixpalmer** [comment]
The tests are taken from the original earcut repo, but modified to match the math.gl code style. In addition, the fixtures are tested to make sure they all work for the new codepath I introduced, namely the optimization that allows passing in precomputed areas for the outline & rings

File: modules/polygon/src/polygon-utils.d.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ type SegmentVisitorFlat = (
 ) => void;
 
 // Segment visitor callback type for polygons defined with array of points.
-type SegmentVisitorPoints = (p1: NumberArray, p2: NumberArray, i1: number, i2: number) => void;
+export type SegmentVisitorPoints = (p1: NumberArray, p2: NumberArray, i1: number, i2: number) => void;
```
Comment (ibgreen) [inline modules/polygon/src/polygon-utils.d.ts:22]: Are we deliberately only exporting the points version, not the flat version?

File: modules/polygon/test/earcut-testdata/expected.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+export default {
```
Comment (ibgreen) [inline modules/polygon/test/earcut-testdata/expected.js:1]: We usually use the following directory structure for test data `modules/polygon/test/data/earcut` (in particular see loaders.gl)

That also allows us to have a README.md in that directory with attribution of the files.

**Review by ibgreen — APPROVED** [review]
Thanks. Assume all changes are non-breaking, but if anything has changed, don't forget to update `upgrade-guide.md`

**Comment by ibgreen** [comment]
@felixpalmer Landing, would be great if you could address the comment in separate PR.

File: modules/polygon/src/polygon-utils.d.ts (hunk 1 above)
Comment (felixpalmer) [reply modules/polygon/src/polygon-utils.d.ts:22]: I needed this type for the `forEachSegment(SegmentVisitorPoints): void;` function so I exported it. The flat version is only used internally AFAIK

File: modules/polygon/test/earcut-testdata/expected.js (hunk 2 above)
Comment (felixpalmer) [reply modules/polygon/test/earcut-testdata/expected.js:1]: Has been addressed in https://github.com/uber-web/math.gl/pull/234

**Pull request merged**